eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -222,6 +252,23 @@ public function countImageReferences($uri)
         return (int) $statement->fetchColumn();
     }
 
+    public function countDistinctImages(): int
+    {
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select(
+                'COUNT(DISTINCT ' . $this->connection->quoteIdentifier('contentobject_attribute_id') .
```

Comment:
Count expression should be taken from an AbstractPlatform implementation (sth like `$this->connection->getDatabasePlatform()->getCountExpression('DISTINCT(contentobject_attribute_id)')`).